src/main/java/com/aws/greengrass/deployment/bootstrap/BootstrapManager.java

```diff
@@ -400,7 +400,9 @@ protected int executeOneBootstrapTask(BootstrapTaskStatus next) throws ServiceUp
             next.setStatus(DONE);
             next.setExitCode(exitCode);
             return exitCode;
-        } catch (InterruptedException | TimeoutException | ServiceLoadException e) {
+        } catch (TimeoutException e) {
+            throw new ServiceUpdateException(e, DeploymentErrorCode.COMPONENT_BOOTSTRAP_TIMEOUT);
+        } catch (InterruptedException | ServiceLoadException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)